### DISCUSSION TITLE
Log scale for histogram x-axis

### DISCUSSION THREAD

**Discussion by Spencer-Comin** [grafana/grafana#38289] [Dashboards, Panels & Visualizations]
**What would you like to be added**:

The graph visualization, when the x-axis is in histogram mode, could have an option to have a logarithmic scale, similar to the y-axis.

![image](https://user-images.githubusercontent.com/52704940/120220879-b5bd5100-c1fa-11eb-9138-986d773422ac.png)

**Why is this needed**:

There is currently no way to show a logarithmic histogram, and it seems more appropriate that it be implemented as a modification to the built-in histogram rather than as a third-party plugin.

**Comment by leeoniya** [comment]
FYI, the histogram mode in the Graph panel (and the panel itself) is deprecated in favor of the new dedicated Histogram panel.

AFAIK, log histograms are fairly uncommon, so i'm not sure it makes sense to support this natively, as they're often awkward to interpret, since they effectively exaggerate the filled areas of the bins.

we have previously discussed whether log histograms are something worth supporting in the core, but have left the question open for now.

e.g.:

![O_6](https://user-images.githubusercontent.com/43234/120222462-97a91e00-c205-11eb-8005-0ea7d29831fa.png)

**Comment by zoltanbedi** [comment]
Hi @Spencer-Comin as Leon said the graph panel is deprecated. Please try to use the new histogram panel. Thank you

**Comment by leeoniya** [comment]
let's leave this open till we figure out what we wanna do

**Comment by Spencer-Comin** [comment]
The use case I am encountering for log histograms is in reporting IO size distribution, where size is generally shown logarithmically (e.g. 0-1kB, 1-2kB, 2-4kB, 4-8kB, etc.), at least in my experience.

**Comment by leeoniya** [comment]
for this use case it seems more appropriate to run some kind of log histogram transform on the raw data (assuming such a transform is made available) and then using it in a BarChart panel with uniform bar widths and text labels (e.g. 0-1kB, 1-2kB, 2-4kB, 4-8kB, etc.)? cc @ryantxu

**Comment by Spencer-Comin** [comment]
This tends to get a bit noisy once the number of buckets exceeds 8 or so. Also the bucket size is rather rigid with this approach. It would be nice to have a histogram where the bucket size and count adapt dynamically to the size of the panel, though this may already be possible with some transform I don't know of yet

**Comment by bruvik** [comment]
> FYI, the histogram mode in the Graph panel (and the panel itself) is deprecated in favor of the new dedicated Histogram panel.

What is this new histogram panel? I can only find references to an old deprecated histogram panel (supporting grafana 4.0)

**Comment by leeoniya** [comment]
https://github.com/grafana/grafana/pull/33752

it's available in 8.0 beta releases since https://github.com/grafana/grafana/pull/34246

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by erezraa** [comment]
Let me add my two cents regarding histograms and log scale(s):
1) The idea to have a log scale is to not allow frequent values to "shadow" low-frequest ones and the "landscape" they represent. 
    For example: a pile of zeros could shadow the rest of the values landscape (which could be showing us an important structure). 
    In the above example one could see the structure by adding a cut: x-axis values > 0 (whatever x is), but this kind of solution might no be easy to implement for every case.
2) Following the above and keeping in mind "being practical": Log scale for Y-axis would probably cover most histogram log scale needs/cases.

**Comment by Ewemek** [comment]
I strongly disagree when someone says that log scale is not a common use case.
Logarithmic scale on X-axis is fairly common in statistical analysis, for instance it's found naturally in concepts such like bernoulli trials.
IMO, allowing such simple option to be chosen for histograms would make Grafana a lot more attractive to people who works with statistics.

**Reply by leeoniya** [reply]
i dont think anyone said log x scale is uncommon; only that log x scale _in histograms_ is uncommon. searching google images for charts of bernoulli trials, i'm not seeing any log x histograms, for example.

**Comment by leeoniya** [comment]
(there is a related discussion about y log scale in histograms)

https://github.com/grafana/grafana/discussions/69756

**Comment by natellium** [comment]
Hey there! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

I will be closing this discussion, but if you’re still interested in this request being implemented please don’t hesitate to open a [feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=&projects=&template=1-feature_requests.md). Sorry for the inconvenience, and thanks for your contributions!

cc @grafana/dataviz-squad
